### PR TITLE
Add mcxn236 usbhs support

### PR DESCRIPTION
**Pull request by NeilChen93**
Add mcxn236 usbhs support

### CODE REVIEW

**Comment by danieldegrasse** [comment]
@NeilChen93 can you take a look at the CI failures?

**Review by danieldegrasse — DISMISSED** [review]

**Comment by dleach02** [comment]
@NeilChen93 this needs a documentation update as well

**Comment by NeilChen93** [comment]
> @NeilChen93 this needs a documentation update as well

**Comment by NeilChen93** [comment]
> @NeilChen93 this needs a documentation update as well
Hi @dleach02, May I ask which document need to be updated?

**Comment by danieldegrasse** [comment]
> > @NeilChen93 this needs a documentation update as well
> > Hi @dleach02, May I ask which document need to be updated?

Add an entry into the hardware features table here: https://github.com/zephyrproject-rtos/zephyr/blob/903b7cf9b62a261abf807d3b0fb9e15cf6497d7f/boards/nxp/frdm_mcxn236/doc/index.rst?plain=1#L80

We should generally do this any time we enable a new feature

**Comment by dleach02** [comment]
When we add features we update the feature list in the doc page.

File: boards/nxp/frdm_mcxn236/doc/index.rst [hunk 1]
```diff
@@ -65,6 +65,8 @@ The FRDM-MCXN236 board configuration supports the following hardware features:
 +-----------+------------+-------------------------------------+
 | ADC       | on-chip    | adc                                 |
 +-----------+------------+-------------------------------------+
+| USBHS     | on-chip    | USB device                          |
++-----------+------------+-------------------------------------+
```
Comment (NeilChen93) [inline boards/nxp/frdm_mcxn236/doc/index.rst:69]: Hi @danieldegrasse , @dleach02 , USB feature already be added in doc index.rst file.

**Comment by mmahadevan108** [comment]
@NeilChen93 , how was this tested.

**Comment by mmahadevan108** [comment]
@NeilChen93, please also update the 4.1 release notes to indicate this support was added.

**Review by danieldegrasse — DISMISSED** [review]

File: boards/nxp/frdm_mcxn236/board.c [hunk 2]
```diff
@@ -8,6 +8,15 @@
...
 #include <fsl_spc.h>
 #include <soc.h>
+#if CONFIG_USB_DC_NXP_EHCI
+#include "usb_phy.h"
+#include "usb.h"
+
+/* USB PHY condfiguration */
+#define BOARD_USB_PHY_D_CAL     (0x04U)
+#define BOARD_USB_PHY_TXCAL45DP (0x07U)
+#define BOARD_USB_PHY_TXCAL45DM (0x07U)
```
Comment (fabiobaltieri) [inline boards/nxp/frdm_mcxn236/board.c:18]: nit: no need for parenthesis on literals

File: boards/nxp/frdm_mcxn236/board.c [hunk 3]
```diff
@@ -206,6 +215,48 @@ static int frdm_mcxn236_init(void)
...
+	 * CORELDO is 1.0V)
+	 */
+	SPC0->ACTIVE_CFG &= ~SPC_ACTIVE_CFG_CORELDO_VDD_DS_MASK;
+	SPC0->ACTIVE_CFG |= SPC_ACTIVE_CFG_DCDC_VDD_LVL(0x3) | SPC_ACTIVE_CFG_CORELDO_VDD_LVL(0x3) |
+			    SPC_ACTIVE_CFG_SYSLDO_VDD_DS_MASK | SPC_ACTIVE_CFG_DCDC_VDD_DS(0x2u);
+	/* Wait until it is done */
+	while (SPC0->SC & SPC_SC_BUSY_MASK) {
+	};
+	if (0u == (SCG0->LDOCSR & SCG_LDOCSR_LDOEN_MASK)) {
+		SCG0->TRIM_LOCK = 0x5a5a0001U;
```
Comment (fabiobaltieri) [inline boards/nxp/frdm_mcxn236/board.c:234]: define for the constant?

File: boards/nxp/frdm_mcxn236/board.c [hunk 4]
```diff
@@ -206,6 +215,48 @@ static int frdm_mcxn236_init(void)
...
+	/* Change the power DCDC to 1.8v (By default, DCDC is 1.8V), CORELDO to 1.1v (By default,
+	 * CORELDO is 1.0V)
+	 */
+	SPC0->ACTIVE_CFG &= ~SPC_ACTIVE_CFG_CORELDO_VDD_DS_MASK;
+	SPC0->ACTIVE_CFG |= SPC_ACTIVE_CFG_DCDC_VDD_LVL(0x3) | SPC_ACTIVE_CFG_CORELDO_VDD_LVL(0x3) |
+			    SPC_ACTIVE_CFG_SYSLDO_VDD_DS_MASK | SPC_ACTIVE_CFG_DCDC_VDD_DS(0x2u);
+	/* Wait until it is done */
+	while (SPC0->SC & SPC_SC_BUSY_MASK) {
+	};
+	if (0u == (SCG0->LDOCSR & SCG_LDOCSR_LDOEN_MASK)) {
```
Comment (fabiobaltieri) [inline boards/nxp/frdm_mcxn236/board.c:233]: these looks backward, most Zephyr code would be `if ((SCG0->LDOCSR & SCG_LDOCSR_LDOEN_MASK) == 0u)`, I suggest changing it, keep it uniform

File: boards/nxp/frdm_mcxn236/board.c [hunk 5]
```diff
@@ -206,6 +215,48 @@ static int frdm_mcxn236_init(void)
...
+		SCG0->TRIM_LOCK = 0x5a5a0001U;
+		SCG0->LDOCSR |= SCG_LDOCSR_LDOEN_MASK;
+		/* wait LDO ready */
+		while (0U == (SCG0->LDOCSR & SCG_LDOCSR_VOUT_OK_MASK)) {
```
Comment (fabiobaltieri) [inline boards/nxp/frdm_mcxn236/board.c:237]: same

File: boards/nxp/frdm_mcxn236/board.c [hunk 6]
```diff
@@ -206,6 +215,48 @@ static int frdm_mcxn236_init(void)
...
+		};
+	}
+	SYSCON->AHBCLKCTRLSET[2] |= SYSCON_AHBCLKCTRL2_USB_HS_MASK |
+				    SYSCON_AHBCLKCTRL2_USB_HS_PHY_MASK;
+	SCG0->SOSCCFG &= ~(SCG_SOSCCFG_RANGE_MASK | SCG_SOSCCFG_EREFS_MASK);
+	/* xtal = 20 ~ 30MHz */
+	SCG0->SOSCCFG = (1U << SCG_SOSCCFG_RANGE_SHIFT) | (1U << SCG_SOSCCFG_EREFS_SHIFT);
```
Comment (fabiobaltieri) [inline boards/nxp/frdm_mcxn236/board.c:244]: you can use `BIT(SCG_SOSCCFG_RANGE_SHIFT) | BIT(SCG_SOSCCFG_EREFS_SHIFT)`

**Review by danieldegrasse — DISMISSED** [review]

**Comment by danieldegrasse** [comment]
Will need a rebase

**Review by danieldegrasse — DISMISSED** [review]

**Review by dleach02 — APPROVED** [review]

**Review by danieldegrasse — APPROVED** [review]

**Pull request merged**